PR: Implement support for Fuji "FLog" and "FGamut" colourspace.

sobotka opened the pull request:
Adds in Fuji F-Log / F-Gamut, without tests.

KelSolaar commented:
Thanks @sobotka ! Will start to take a look at that. Seems like the tests are breaking on the reciprocity case.

KelSolaar commented on colour/__init__.py:
```diff
@@ -908,6 +908,10 @@ def __getattr__(self, attribute):
             'colour.models.log_decoding_VLog',
         ],
+        [
+            'colour.log_decoding_FLog',
```
Those API changes are not required here because the definitions have never been renamed or their access point changed: it is the first time they are featured in Colour.

KelSolaar commented on colour/__init__.py:
```diff
@@ -988,6 +992,10 @@ def __getattr__(self, attribute):
             'colour.models.log_encoding_VLog',
         ],
+        [
+            'colour.log_encoding_FLog',
```
Same here, not needed.

KelSolaar commented on colour/models/rgb/dataset/f_gamut.py:
```diff
@@ -0,0 +1,102 @@
+
+References
+----------
+-   :cite:`FujiVer1` : F-Log Data Sheet Ver.1.0.
```
I can change it but if you want the citation which depends on the entry I added to Mendeley (and will push to the *BIBLIOGRAPHY.bib* file) should be like that:

> -   :cite:`Fujifilm2016` : Fujifilm. (2016). F-Log Data Sheet Ver.1.0. Retrieved from https://www.fujifilm.com/support/digital_cameras/software/lut/pdf/F-Log_DataSheet_E_Ver.1.0.pdf

KelSolaar commented on colour/models/rgb/transfer_functions/fuji_flog.py:
```diff
@@ -0,0 +1,201 @@
+# -*- coding: utf-8 -*-
+"""
+Fuji F-Log Log Encoding
+============================
```
Might need to shave a few `=`, in that direction it is no big deal but if shorter it slightly borks the documentation :)

KelSolaar commented on colour/models/rgb/transfer_functions/fuji_flog.py:
```diff
@@ -0,0 +1,201 @@
+
+    References
+    ----------
+    :cite:`FujiVer1`
```
The citation keys will also need update.

KelSolaar commented on colour/models/rgb/transfer_functions/fuji_flog.py:
```diff
@@ -0,0 +1,201 @@
+    Examples
+    --------
+    >>> log_encoding_FLog(0.18)  # doctest: +ELLIPSIS
+    0.46333651051465585...
```
Should trim at 7 decimals, the ellipsis is here to help platform differences on numerical computations.

> 0.4633365...

KelSolaar commented on colour/models/rgb/transfer_functions/fuji_flog.py:
```diff
@@ -0,0 +1,201 @@
+    Examples
+    --------
+    >>> log_decoding_FLog(0.46333651051465585)  # doctest: +ELLIPSIS
+    0.18000000000000002...
```
Likewise 7 decimals :)

KelSolaar commented on colour/models/rgb/transfer_functions/fuji_flog.py:
```diff
@@ -0,0 +1,201 @@
+def log_encoding_FLog(L_in,
+                      bit_depth=10,
+                      out_legal=False,
+                      in_reflection=True,
```
Do we need support for that? Seems like the reference does not specifically mention it.

KelSolaar commented:
@sobotka : Thanks again, I added a few minor comments that should be quick to address. I will need to spend some time to understand why the current unit tests are breaking.

Two things:

- Would be great to squash the *Fix typo on F_out* and *Update doctest string* commits into *Add F-Log and F-Gamut* when we are done to keep the history a bit tidy.
- We will need to add a `colour/models/rgb/transfer_functions/tests/test_fuji_flog.py` unit test module.

@nick-shaw : Do you think we should drop the `in_reflection` argument?

sobotka commented:
Rebased against develop's head and squashed everything into a single merge point.

I believe you can change the branch from [sobotka:fuji](/sobotka/colour/tree/fuji) to [sobotka:fuji-revisions](/sobotka/colour/tree/fuji-revisions)?

KelSolaar commented:
I will close this one and use #480 instead!
